packages/salesforcedx-vscode-core/src/commands/util/parameterGatherers.ts

```diff
@@ -161,6 +165,88 @@ export class DemoModePromptGatherer implements ParametersGatherer<{}> {
   }
 }
 
+export class SelectLwcComponentDir
+  implements ParametersGatherer<{ fileName: string; outputdir: string }> {
+  private typeDir: string;
+  private typeDirRequired: boolean | undefined;
+  public static readonly defaultOutput = path.join('main', 'default');
+
+  public constructor(typeDir: string, typeDirRequired?: boolean) {
+    this.typeDir = typeDir;
+    this.typeDirRequired = typeDirRequired;
+  }
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string; outputdir: string }>
+  > {
+    let packageDirs: string[] = [];
+    try {
+      packageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    } catch (e) {
+      if (
+        e.name !== 'NoPackageDirectoryPathsFound' &&
+        e.name !== 'NoPackageDirectoriesFound'
+      ) {
+        throw e;
+      }
+    }
+
+    const dirOptions = this.getDefaultOptions(packageDirs);
+    let outputdir = await this.showMenu(dirOptions);
+    const namePathMap = new Map();
+    let filepath;
+    let filename;
+    let fileName;
+
+    if (outputdir) {
+      const fullPath = path.join(getRootWorkspacePath(), outputdir);
+      const registry = new RegistryAccess();
+      const components = registry.getComponentsFromPath(fullPath);
+      const lwccomponents = components.filter(
+        lwc =>
+          lwc.type.name === registryData.types.lightningcomponentbundle.name
+      );
+      const lwcNames = lwccomponents.map(lwc => lwc.fullName);
+
+      for (const component of lwccomponents) {
+        namePathMap.set(component.fullName, component.xml);
+      }
+
+      filename = await this.showLwcMenu(lwcNames);
+      filepath = namePathMap.get(filename);
+      filepath = filepath.replace('-meta.xml', '');
+      fileName = path.basename(filepath, '.js');
+      outputdir = path.join(outputdir, fileName);
+    }
+
+    return outputdir && fileName
+      ? {
+          type: 'CONTINUE',
+          data: { fileName, outputdir }
+        }
+      : { type: 'CANCEL' };
+  }
+
+  public getDefaultOptions(packageDirectories: string[]): string[] {
+    const options = packageDirectories.map(packageDir =>
+      path.join(packageDir, SelectLwcComponentDir.defaultOutput, this.typeDir)
+    );
+    return options;
+  }
+
+  public async showMenu(options: string[]): Promise<string | undefined> {
+    return await vscode.window.showQuickPick(options, {
+      placeHolder: nls.localize('parameter_gatherer_enter_dir_name')
+    } as vscode.QuickPickOptions);
+  }
+
+  public async showLwcMenu(options: string[]): Promise<string | undefined> {
+    return await vscode.window.showQuickPick(options, {
+      placeHolder: nls.localize('parameter_gatherer_enter_lwc_name')
+    } as vscode.QuickPickOptions);
+  }
+}
```

Comment:
We can consolidate these into one since it's the same functionality just a difference `placeHolder` message